# Importing the Dataset

In [38]:
# importing the dataset as a pandas dataframe
import pandas as pd
df = pd.read_csv('task.csv')

print(df.shape)
print(df.columns)
df.head(3)

(80, 14)
Index(['Timestamp', 'Name', 'Year in School', 'Major',
       'Second Major (if applicable)', 'Minor (if applicable)',
       'Second Minor (if applicable)', 'GPA', 'Which team interests you?',
       'Why does this team interest you?',
       'How much time can you commit per week?',
       'What value will you bring to Quant?',
       'What do you hope to get out of Quant?',
       '1 - no, 2 - maybe, 3 - yes, 4 - intern'],
      dtype='object')


Timestamp            Name Year in School  \
0  12/2/2021 17:41:15   Jessica Smith       Freshman   
1  12/2/2021 17:51:17  Teresa Aguilar       Freshman   
2  12/2/2021 18:00:50  Angela Miranda      Sophomore   

                            Major Second Major (if applicable)  \
0                Computer Science                  Mathematics   
1                Computer Science                          NaN   
2  Mathematics & Computer Science                          NaN   

  Minor (if applicable) Second Minor (if applicable)   GPA  \
0        Urban Planning                          NaN     4   
1           Mathematics                          NaN     4   
2                   NaN                          NaN  3.91   

  Which team interests you?  \
0     Quantitative Research   
1      Software Development   
2      Software Development   

                    Why does this team interest you?  \
0  I am interested in both Quant Research and Sof...   
1  I'm interested in both the Software Developmen...   
2  The Software Development team interests me bec...   

              How much time can you commit per week?  \
0  5 to 10 hours per week, but up to 20 depending...   
1                                        10-15 hours   
2                                        10-15 hours   

                 What value will you bring to Quant?  \
0  I have extensive experience working on enginee...   
1  The main value I'd bring to quant would be fro...   
2  Aside from taking most of the core CS classes ...   

               What do you hope to get out of Quant?  \
0  It is extremely difficult to learn about quant...   
1  I'm hoping to gain more experience with quanti...   
2  I'd like to learn more about system programmin...   

   1 - no, 2 - maybe, 3 - yes, 4 - intern  
0                                     3.0  
1                                     3.0  
2                                     4.0

# Exploratory Data Analysis and Data Cleaning



In [39]:
import matplotlib.pyplot as plt
import seaborn as sns
df.nunique(axis=0)


Timestamp                                 80
Name                                      79
Year in School                             6
Major                                     36
Second Major (if applicable)               8
Minor (if applicable)                     15
Second Minor (if applicable)               5
GPA                                       38
Which team interests you?                 12
Why does this team interest you?          80
How much time can you commit per week?    63
What value will you bring to Quant?       80
What do you hope to get out of Quant?     80
1 - no, 2 - maybe, 3 - yes, 4 - intern     4
dtype: int64

In [40]:
df.describe()

1 - no, 2 - maybe, 3 - yes, 4 - intern
count                               60.000000
mean                                 2.583333
std                                  0.787437
min                                  1.000000
25%                                  2.000000
50%                                  3.000000
75%                                  3.000000
max                                  4.000000

only one column is numeric data, so we must convert the others as necessary

## GPA Column

In [41]:
# convert gpa column to floats


df['GPA'] = pd.to_numeric(df['GPA'],errors = 'coerce')

df2 = df.dropna(subset=['GPA'])
print((abs(len(df2)-len(df)))/len(df)*100) # percentage of missing values in dataset, 




18.75


In [42]:
df.corr()

GPA  \
GPA                                     1.000000   
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.553592   

                                        1 - no, 2 - maybe, 3 - yes, 4 - intern  
GPA                                                                   0.553592  
1 - no, 2 - maybe, 3 - yes, 4 - intern                                1.000000

Since 18.75% is a reasonable margin for error we can perform some sort of imputation to the NaN values. 


In [43]:
from sklearn.impute import SimpleImputer
import numpy as np
imp = SimpleImputer(missing_values=np.nan, strategy = 'mean')
imp = imp.fit(df[['GPA']])
df['GPA'] = imp.transform(df[['GPA']]).ravel()
df2 = df.dropna(subset= ['GPA'])
print((abs(len(df2)-len(df)))/len(df)*100) # percentage of missing values in dataset, 
# Goal is to check that all nan values have been replaced by the mean value in the datset. 


0.0


Note: Another possible way to do imputation is K-nearest neighbours but it seems more intuitive to use only the gpa column's data for imputation as we don't have any variables that are clearly correlated with GPA. 

## Hours Available

In [44]:
df['hours']  = df['How much time can you commit per week?'] #making column easier to work with having such a wordy name is annoying
df = df.assign(hours = lambda x: x['hours'].str.extract('(\d+)')) # using regex to scrape out first number from the range/ single number each person gave. 
print(df['hours'])


0      5
1     10
2     10
3     10
4      3
      ..
75    15
76     2
77     5
78     6
79    10
Name: hours, Length: 80, dtype: object


In [45]:
# since dtype is still object, we should convert it to numerical data again for ease of use. 
df['hours'] = pd.to_numeric(df['hours'],errors = 'coerce')
df['hours']


0      5.0
1     10.0
2     10.0
3     10.0
4      3.0
      ... 
75    15.0
76     2.0
77     5.0
78     6.0
79    10.0
Name: hours, Length: 80, dtype: float64

In [46]:
df2 = df.dropna(subset=['hours'])
print((abs(len(df2)-len(df)))/len(df)*100) # percent of hours values that are na. 

3.75


As we can see only 3.75% of the values for the hours are NaN, therefore the column is very usable and should probably be imputed.

In [47]:
imp = SimpleImputer(missing_values=np.nan, strategy = 'mean')
imp = imp.fit(df[['hours']])
df['hours'] = imp.transform(df[['hours']]).ravel()
df2 = df.dropna(subset= ['hours'])
print((abs(len(df2)-len(df)))/len(df)*100) # percentage of missing values in dataset, 
# Goal is to check that all nan values have been replaced by the mean value in the datset. 

0.0


Above code is very similar to the procedure for the gpa dataset, replaced all NaN values with the average GPA in the dataset.

In [48]:
df.drop(columns = ['How much time can you commit per week?']) # irrelevant now that we have the hours column

Timestamp             Name Year in School  \
0    12/2/2021 17:41:15    Jessica Smith       Freshman   
1    12/2/2021 17:51:17   Teresa Aguilar       Freshman   
2    12/2/2021 18:00:50   Angela Miranda      Sophomore   
3    12/2/2021 19:00:07  Andre Mccormick         Junior   
4    12/2/2021 21:23:32   Marcus Gilbert         Junior   
..                  ...              ...            ...   
75  12/20/2021 18:10:45       Jose Jones      Sophomore   
76  12/20/2021 21:38:28  Elizabeth Wolfe      Sophomore   
77   12/21/2021 0:26:20   Colleen Gordon       Master's   
78   12/21/2021 1:39:35        Rita Vega      Sophomore   
79  12/21/2021 10:02:09    Justin Martin         Junior   

                             Major Second Major (if applicable)  \
0                 Computer Science                  Mathematics   
1                 Computer Science                          NaN   
2   Mathematics & Computer Science                          NaN   
3    Statistics & Computer Science                          NaN   
4                       Statistics                          NaN   
..                             ...                          ...   
75         Finance in Agribusiness                          NaN   
76                         Finance          Information Systems   
77             Information science                          NaN   
78                Computer Science                          NaN   
79       Math and Computer Science                          NaN   

   Minor (if applicable)         Second Minor (if applicable)   GPA  \
0         Urban Planning                                  NaN  4.00   
1            Mathematics                                  NaN  4.00   
2                    NaN                                  NaN  3.91   
3            Mathematics                                  NaN  3.74   
4                    NaN                                  NaN  3.92   
..                   ...                                  ...   ...   
75                   NaN                                  NaN  3.10   
76                   NaN                                  NaN  4.00   
77                   NaN                                  NaN  3.90   
78            Statistics  Hopefully Technology and management  3.86   
79                   NaN                                  NaN  3.94   

                            Which team interests you?  \
0                               Quantitative Research   
1                                Software Development   
2                                Software Development   
3   Software Development, Strategy Implementation,...   
4                               Quantitative Research   
..                                                ...   
75  Strategy Implementation, Quantitative Research...   
76  Software Development, Strategy Implementation,...   
77                            Strategy Implementation   
78                               Software Development   
79                               Software Development   

                     Why does this team interest you?  \
0   I am interested in both Quant Research and Sof...   
1   I'm interested in both the Software Developmen...   
2   The Software Development team interests me bec...   
3        Interested in drawing insights from markets.   
4   I want to pursue a career in the field of quan...   
..                                                ...   
75  I want to be a part of something bigger than m...   
76  I have been interested in tech, and investing ...   
77  My experience in data analytics can be helpful...   
78  This team interests me because I am very inter...   
79         I enjoy writing code to build new products   

                  What value will you bring to Quant?  \
0   I have extensive experience working on enginee...   
1   The main value I'd bring to quant would be fro...   
2   Aside from taking most of the core CS classes ...   
3   My technical skills and background will allow

In [49]:
df.corr()

GPA  \
GPA                                     1.000000   
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.512905   
hours                                  -0.198078   

                                        1 - no, 2 - maybe, 3 - yes, 4 - intern  \
GPA                                                                   0.512905   
1 - no, 2 - maybe, 3 - yes, 4 - intern                                1.000000   
hours                                                                 0.020963   

                                           hours  
GPA                                    -0.198078  
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.020963  
hours                                   1.000000

Now that we've handled most of our numerical data and can see that there is some correlation between the GPA and the score of the candidate, we'll try and analyze some of the text based data. 


### Text Based Data

In [50]:
df['word count'] = (df['Why does this team interest you?'] + df['What value will you bring to Quant?'] + df['What do you hope to get out of Quant?']).apply(lambda x: len(str(x).split(" ")))
df.head()

Timestamp             Name Year in School  \
0  12/2/2021 17:41:15    Jessica Smith       Freshman   
1  12/2/2021 17:51:17   Teresa Aguilar       Freshman   
2  12/2/2021 18:00:50   Angela Miranda      Sophomore   
3  12/2/2021 19:00:07  Andre Mccormick         Junior   
4  12/2/2021 21:23:32   Marcus Gilbert         Junior   

                            Major Second Major (if applicable)  \
0                Computer Science                  Mathematics   
1                Computer Science                          NaN   
2  Mathematics & Computer Science                          NaN   
3   Statistics & Computer Science                          NaN   
4                      Statistics                          NaN   

  Minor (if applicable) Second Minor (if applicable)   GPA  \
0        Urban Planning                          NaN  4.00   
1           Mathematics                          NaN  4.00   
2                   NaN                          NaN  3.91   
3           Mathematics                          NaN  3.74   
4                   NaN                          NaN  3.92   

                           Which team interests you?  \
0                              Quantitative Research   
1                               Software Development   
2                               Software Development   
3  Software Development, Strategy Implementation,...   
4                              Quantitative Research   

                    Why does this team interest you?  \
0  I am interested in both Quant Research and Sof...   
1  I'm interested in both the Software Developmen...   
2  The Software Development team interests me bec...   
3       Interested in drawing insights from markets.   
4  I want to pursue a career in the field of quan...   

              How much time can you commit per week?  \
0  5 to 10 hours per week, but up to 20 depending...   
1                                        10-15 hours   
2                                        10-15 hours   
3                                           10 hours   
4                                          3–5 hours   

                 What value will you bring to Quant?  \
0  I have extensive experience working on enginee...   
1  The main value I'd bring to quant would be fro...   
2  Aside from taking most of the core CS classes ...   
3  My technical skills and background will allow ...   
4  I believe my knowledge in the field of Statist...   

               What do you hope to get out of Quant?  \
0  It is extremely difficult to learn about quant...   
1  I'm hoping to gain more experience with quanti...   
2  I'd like to learn more about system programmin...   
3  Learn new skills and most of all, develop a ne...   
4  Some project experience and collaboration skills.   

   1 - no, 2 - maybe, 3 - yes, 4 - intern  hours  word count  
0                                     3.0    5.0         122  
1                                     3.0   10.0         166  
2                                     4.0   10.0         241  
3                                     1.0   10.0          46  
4                                     2.0    3.0          68

In [51]:
df.corr()

GPA  \
GPA                                     1.000000   
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.512905   
hours                                  -0.198078   
word count                              0.101102   

                                        1 - no, 2 - maybe, 3 - yes, 4 - intern  \
GPA                                                                   0.512905   
1 - no, 2 - maybe, 3 - yes, 4 - intern                                1.000000   
hours                                                                 0.020963   
word count                                                            0.339824   

                                           hours  word count  
GPA                                    -0.198078    0.101102  
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.020963    0.339824  
hours                                   1.000000   -0.122676  
word count                             -0.122676    1.000000

In [52]:
df['char count'] = (df['Why does this team interest you?'] + df['What value will you bring to Quant?'] + df['What do you hope to get out of Quant?']).str.len()
df.head()

Timestamp             Name Year in School  \
0  12/2/2021 17:41:15    Jessica Smith       Freshman   
1  12/2/2021 17:51:17   Teresa Aguilar       Freshman   
2  12/2/2021 18:00:50   Angela Miranda      Sophomore   
3  12/2/2021 19:00:07  Andre Mccormick         Junior   
4  12/2/2021 21:23:32   Marcus Gilbert         Junior   

                            Major Second Major (if applicable)  \
0                Computer Science                  Mathematics   
1                Computer Science                          NaN   
2  Mathematics & Computer Science                          NaN   
3   Statistics & Computer Science                          NaN   
4                      Statistics                          NaN   

  Minor (if applicable) Second Minor (if applicable)   GPA  \
0        Urban Planning                          NaN  4.00   
1           Mathematics                          NaN  4.00   
2                   NaN                          NaN  3.91   
3           Mathematics                          NaN  3.74   
4                   NaN                          NaN  3.92   

                           Which team interests you?  \
0                              Quantitative Research   
1                               Software Development   
2                               Software Development   
3  Software Development, Strategy Implementation,...   
4                              Quantitative Research   

                    Why does this team interest you?  \
0  I am interested in both Quant Research and Sof...   
1  I'm interested in both the Software Developmen...   
2  The Software Development team interests me bec...   
3       Interested in drawing insights from markets.   
4  I want to pursue a career in the field of quan...   

              How much time can you commit per week?  \
0  5 to 10 hours per week, but up to 20 depending...   
1                                        10-15 hours   
2                                        10-15 hours   
3                                           10 hours   
4                                          3–5 hours   

                 What value will you bring to Quant?  \
0  I have extensive experience working on enginee...   
1  The main value I'd bring to quant would be fro...   
2  Aside from taking most of the core CS classes ...   
3  My technical skills and background will allow ...   
4  I believe my knowledge in the field of Statist...   

               What do you hope to get out of Quant?  \
0  It is extremely difficult to learn about quant...   
1  I'm hoping to gain more experience with quanti...   
2  I'd like to learn more about system programmin...   
3  Learn new skills and most of all, develop a ne...   
4  Some project experience and collaboration skills.   

   1 - no, 2 - maybe, 3 - yes, 4 - intern  hours  word count  char count  
0                                     3.0    5.0         122         790  
1                                     3.0   10.0         166        1081  
2                                     4.0   10.0         241        1460  
3                                     1.0   10.0          46         303  
4                                     2.0    3.0          68         384

In [53]:
df.corr()

GPA  \
GPA                                     1.000000   
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.512905   
hours                                  -0.198078   
word count                              0.101102   
char count                              0.110002   

                                        1 - no, 2 - maybe, 3 - yes, 4 - intern  \
GPA                                                                   0.512905   
1 - no, 2 - maybe, 3 - yes, 4 - intern                                1.000000   
hours                                                                 0.020963   
word count                                                            0.339824   
char count                                                            0.335874   

                                           hours  word count  char count  
GPA                                    -0.198078    0.101102    0.110002  
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.020963    0.339824    0.335874  
hours                                   1.000000   -0.122676   -0.113463  
word count                             -0.122676    1.000000    0.994536  
char count                             -0.113463    0.994536    1.000000

In [54]:
def avg_word(sentence):
  words = sentence.split()
  return (sum(len(word) for word in words)/len(words))
df['avg_word'] = (df['Why does this team interest you?'] + df['What value will you bring to Quant?'] + df['What do you hope to get out of Quant?']).apply(lambda x: avg_word(x))
df.head()

Timestamp             Name Year in School  \
0  12/2/2021 17:41:15    Jessica Smith       Freshman   
1  12/2/2021 17:51:17   Teresa Aguilar       Freshman   
2  12/2/2021 18:00:50   Angela Miranda      Sophomore   
3  12/2/2021 19:00:07  Andre Mccormick         Junior   
4  12/2/2021 21:23:32   Marcus Gilbert         Junior   

                            Major Second Major (if applicable)  \
0                Computer Science                  Mathematics   
1                Computer Science                          NaN   
2  Mathematics & Computer Science                          NaN   
3   Statistics & Computer Science                          NaN   
4                      Statistics                          NaN   

  Minor (if applicable) Second Minor (if applicable)   GPA  \
0        Urban Planning                          NaN  4.00   
1           Mathematics                          NaN  4.00   
2                   NaN                          NaN  3.91   
3           Mathematics                          NaN  3.74   
4                   NaN                          NaN  3.92   

                           Which team interests you?  \
0                              Quantitative Research   
1                               Software Development   
2                               Software Development   
3  Software Development, Strategy Implementation,...   
4                              Quantitative Research   

                    Why does this team interest you?  \
0  I am interested in both Quant Research and Sof...   
1  I'm interested in both the Software Developmen...   
2  The Software Development team interests me bec...   
3       Interested in drawing insights from markets.   
4  I want to pursue a career in the field of quan...   

              How much time can you commit per week?  \
0  5 to 10 hours per week, but up to 20 depending...   
1                                        10-15 hours   
2                                        10-15 hours   
3                                           10 hours   
4                                          3–5 hours   

                 What value will you bring to Quant?  \
0  I have extensive experience working on enginee...   
1  The main value I'd bring to quant would be fro...   
2  Aside from taking most of the core CS classes ...   
3  My technical skills and background will allow ...   
4  I believe my knowledge in the field of Statist...   

               What do you hope to get out of Quant?  \
0  It is extremely difficult to learn about quant...   
1  I'm hoping to gain more experience with quanti...   
2  I'd like to learn more about system programmin...   
3  Learn new skills and most of all, develop a ne...   
4  Some project experience and collaboration skills.   

   1 - no, 2 - maybe, 3 - yes, 4 - intern  hours  word count  char count  \
0                                     3.0    5.0         122         790   
1                                     3.0   10.0         166        1081   
2                                     4.0   10.0         241        1460   
3                                     1.0   10.0          46         303   
4                                     2.0    3.0          68         384   

   avg_word  
0  5.483607  
1  5.518072  
2  5.062241  
3  5.608696  
4  4.661765

In [55]:
df.corr()

GPA  \
GPA                                     1.000000   
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.512905   
hours                                  -0.198078   
word count                              0.101102   
char count                              0.110002   
avg_word                                0.134431   

                                        1 - no, 2 - maybe, 3 - yes, 4 - intern  \
GPA                                                                   0.512905   
1 - no, 2 - maybe, 3 - yes, 4 - intern                                1.000000   
hours                                                                 0.020963   
word count                                                            0.339824   
char count                                                            0.335874   
avg_word                                                             -0.141008   

                                           hours  word count  char count  \
GPA                                    -0.198078    0.101102    0.110002   
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.020963    0.339824    0.335874   
hours                                   1.000000   -0.122676   -0.113463   
word count                             -0.122676    1.000000    0.994536   
char count                             -0.113463    0.994536    1.000000   
avg_word                                0.030919   -0.117322   -0.042036   

                                        avg_word  
GPA                                     0.134431  
1 - no, 2 - maybe, 3 - yes, 4 - intern -0.141008  
hours                                   0.030919  
word count                             -0.117322  
char count                             -0.042036  
avg_word                                1.000000

In [56]:
from nltk.corpus import stopwords
stop = stopwords.words('english')
df['stopwords'] = (df['Why does this team interest you?'] + df['What value will you bring to Quant?'] + df['What do you hope to get out of Quant?']).apply(lambda x: len([x for x in x.split() if x in stop]))
df.head()

Timestamp             Name Year in School  \
0  12/2/2021 17:41:15    Jessica Smith       Freshman   
1  12/2/2021 17:51:17   Teresa Aguilar       Freshman   
2  12/2/2021 18:00:50   Angela Miranda      Sophomore   
3  12/2/2021 19:00:07  Andre Mccormick         Junior   
4  12/2/2021 21:23:32   Marcus Gilbert         Junior   

                            Major Second Major (if applicable)  \
0                Computer Science                  Mathematics   
1                Computer Science                          NaN   
2  Mathematics & Computer Science                          NaN   
3   Statistics & Computer Science                          NaN   
4                      Statistics                          NaN   

  Minor (if applicable) Second Minor (if applicable)   GPA  \
0        Urban Planning                          NaN  4.00   
1           Mathematics                          NaN  4.00   
2                   NaN                          NaN  3.91   
3           Mathematics                          NaN  3.74   
4                   NaN                          NaN  3.92   

                           Which team interests you?  \
0                              Quantitative Research   
1                               Software Development   
2                               Software Development   
3  Software Development, Strategy Implementation,...   
4                              Quantitative Research   

                    Why does this team interest you?  \
0  I am interested in both Quant Research and Sof...   
1  I'm interested in both the Software Developmen...   
2  The Software Development team interests me bec...   
3       Interested in drawing insights from markets.   
4  I want to pursue a career in the field of quan...   

              How much time can you commit per week?  \
0  5 to 10 hours per week, but up to 20 depending...   
1                                        10-15 hours   
2                                        10-15 hours   
3                                           10 hours   
4                                          3–5 hours   

                 What value will you bring to Quant?  \
0  I have extensive experience working on enginee...   
1  The main value I'd bring to quant would be fro...   
2  Aside from taking most of the core CS classes ...   
3  My technical skills and background will allow ...   
4  I believe my knowledge in the field of Statist...   

               What do you hope to get out of Quant?  \
0  It is extremely difficult to learn about quant...   
1  I'm hoping to gain more experience with quanti...   
2  I'd like to learn more about system programmin...   
3  Learn new skills and most of all, develop a ne...   
4  Some project experience and collaboration skills.   

   1 - no, 2 - maybe, 3 - yes, 4 - intern  hours  word count  char count  \
0                                     3.0    5.0         122         790   
1                                     3.0   10.0         166        1081   
2                                     4.0   10.0         241        1460   
3                                     1.0   10.0          46         303   
4                                     2.0    3.0          68         384   

   avg_word  stopwords  
0  5.483607         48  
1  5.518072         68  
2  5.062241        104  
3  5.608696         18  
4  4.661765         28

In [57]:
df.corr()

GPA  \
GPA                                     1.000000   
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.512905   
hours                                  -0.198078   
word count                              0.101102   
char count                              0.110002   
avg_word                                0.134431   
stopwords                               0.092039   

                                        1 - no, 2 - maybe, 3 - yes, 4 - intern  \
GPA                                                                   0.512905   
1 - no, 2 - maybe, 3 - yes, 4 - intern                                1.000000   
hours                                                                 0.020963   
word count                                                            0.339824   
char count                                                            0.335874   
avg_word                                                             -0.141008   
stopwords                                                             0.324417   

                                           hours  word count  char count  \
GPA                                    -0.198078    0.101102    0.110002   
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.020963    0.339824    0.335874   
hours                                   1.000000   -0.122676   -0.113463   
word count                             -0.122676    1.000000    0.994536   
char count                             -0.113463    0.994536    1.000000   
avg_word                                0.030919   -0.117322   -0.042036   
stopwords                              -0.130189    0.992611    0.980561   

                                        avg_word  stopwords  
GPA                                     0.134431   0.092039  
1 - no, 2 - maybe, 3 - yes, 4 - intern -0.141008   0.324417  
hours                                   0.030919  -0.130189  
word count                             -0.117322   0.992611  
char count                             -0.042036   0.980561  
avg_word                                1.000000  -0.163788  
stopwords                              -0.163788   1.000000

In [58]:
df['Which team interests you?'] = df['Which team interests you?'].str.split(", ")

In [60]:
def job_type(value):
    l = []
    if 'Quantitative Research' in value:
        l.append(1)
    if 'Software Development' in value:
        l.append(2)
    if 'Strategy Implementation' in value:
        l.append(3)
    if 'Business' in value:
        l.append(4)
    return l


df['team numbers'] = (df['Which team interests you?']).apply(lambda x: job_type(x))
df.head()


Timestamp             Name Year in School  \
0  12/2/2021 17:41:15    Jessica Smith       Freshman   
1  12/2/2021 17:51:17   Teresa Aguilar       Freshman   
2  12/2/2021 18:00:50   Angela Miranda      Sophomore   
3  12/2/2021 19:00:07  Andre Mccormick         Junior   
4  12/2/2021 21:23:32   Marcus Gilbert         Junior   

                            Major Second Major (if applicable)  \
0                Computer Science                  Mathematics   
1                Computer Science                          NaN   
2  Mathematics & Computer Science                          NaN   
3   Statistics & Computer Science                          NaN   
4                      Statistics                          NaN   

  Minor (if applicable) Second Minor (if applicable)   GPA  \
0        Urban Planning                          NaN  4.00   
1           Mathematics                          NaN  4.00   
2                   NaN                          NaN  3.91   
3           Mathematics                          NaN  3.74   
4                   NaN                          NaN  3.92   

                           Which team interests you?  \
0                            [Quantitative Research]   
1                             [Software Development]   
2                             [Software Development]   
3  [Software Development, Strategy Implementation...   
4                            [Quantitative Research]   

                    Why does this team interest you?  \
0  I am interested in both Quant Research and Sof...   
1  I'm interested in both the Software Developmen...   
2  The Software Development team interests me bec...   
3       Interested in drawing insights from markets.   
4  I want to pursue a career in the field of quan...   

              How much time can you commit per week?  \
0  5 to 10 hours per week, but up to 20 depending...   
1                                        10-15 hours   
2                                        10-15 hours   
3                                           10 hours   
4                                          3–5 hours   

                 What value will you bring to Quant?  \
0  I have extensive experience working on enginee...   
1  The main value I'd bring to quant would be fro...   
2  Aside from taking most of the core CS classes ...   
3  My technical skills and background will allow ...   
4  I believe my knowledge in the field of Statist...   

               What do you hope to get out of Quant?  \
0  It is extremely difficult to learn about quant...   
1  I'm hoping to gain more experience with quanti...   
2  I'd like to learn more about system programmin...   
3  Learn new skills and most of all, develop a ne...   
4  Some project experience and collaboration skills.   

   1 - no, 2 - maybe, 3 - yes, 4 - intern  hours  word count  char count  \
0                                     3.0    5.0         122         790   
1                                     3.0   10.0         166        1081   
2                                     4.0   10.0         241        1460   
3                                     1.0   10.0          46         303   
4                                     2.0    3.0          68         384   

   avg_word  stopwords team numbers  
0  5.483607         48          [1]  
1  5.518072         68          [2]  
2  5.062241        104          [2]  
3  5.608696         18    [1, 2, 3]  
4  4.661765         28          [1]

In [61]:
def class_type(value):
    if value == 'Freshman':
        return 1
    if value == 'Sophomore':
        return 2
    if value == 'Junior':
        return 3
    if value == 'Senior':
        return 4
    return 0
df['class value'] = df['Year in School'].apply(lambda x: class_type(x))
df.head()

Timestamp             Name Year in School  \
0  12/2/2021 17:41:15    Jessica Smith       Freshman   
1  12/2/2021 17:51:17   Teresa Aguilar       Freshman   
2  12/2/2021 18:00:50   Angela Miranda      Sophomore   
3  12/2/2021 19:00:07  Andre Mccormick         Junior   
4  12/2/2021 21:23:32   Marcus Gilbert         Junior   

                            Major Second Major (if applicable)  \
0                Computer Science                  Mathematics   
1                Computer Science                          NaN   
2  Mathematics & Computer Science                          NaN   
3   Statistics & Computer Science                          NaN   
4                      Statistics                          NaN   

  Minor (if applicable) Second Minor (if applicable)   GPA  \
0        Urban Planning                          NaN  4.00   
1           Mathematics                          NaN  4.00   
2                   NaN                          NaN  3.91   
3           Mathematics                          NaN  3.74   
4                   NaN                          NaN  3.92   

                           Which team interests you?  \
0                            [Quantitative Research]   
1                             [Software Development]   
2                             [Software Development]   
3  [Software Development, Strategy Implementation...   
4                            [Quantitative Research]   

                    Why does this team interest you?  ...  \
0  I am interested in both Quant Research and Sof...  ...   
1  I'm interested in both the Software Developmen...  ...   
2  The Software Development team interests me bec...  ...   
3       Interested in drawing insights from markets.  ...   
4  I want to pursue a career in the field of quan...  ...   

                 What value will you bring to Quant?  \
0  I have extensive experience working on enginee...   
1  The main value I'd bring to quant would be fro...   
2  Aside from taking most of the core CS classes ...   
3  My technical skills and background will allow ...   
4  I believe my knowledge in the field of Statist...   

               What do you hope to get out of Quant?  \
0  It is extremely difficult to learn about quant...   
1  I'm hoping to gain more experience with quanti...   
2  I'd like to learn more about system programmin...   
3  Learn new skills and most of all, develop a ne...   
4  Some project experience and collaboration skills.   

  1 - no, 2 - maybe, 3 - yes, 4 - intern  hours  word count  char count  \
0                                    3.0    5.0         122         790   
1                                    3.0   10.0         166        1081   
2                                    4.0   10.0         241        1460   
3                                    1.0   10.0          46         303   
4                                    2.0    3.0          68         384   

   avg_word  stopwords  team numbers class value  
0  5.483607         48           [1]           1  
1  5.518072         68           [2]           1  
2  5.062241        104           [2]           2  
3  5.608696         18     [1, 2, 3]           3  
4  4.661765         28           [1]           3  

[5 rows x 21 columns]

In [62]:
df.corr()

GPA  \
GPA                                     1.000000   
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.512905   
hours                                  -0.198078   
word count                              0.101102   
char count                              0.110002   
avg_word                                0.134431   
stopwords                               0.092039   
class value                            -0.156800   

                                        1 - no, 2 - maybe, 3 - yes, 4 - intern  \
GPA                                                                   0.512905   
1 - no, 2 - maybe, 3 - yes, 4 - intern                                1.000000   
hours                                                                 0.020963   
word count                                                            0.339824   
char count                                                            0.335874   
avg_word                                                             -0.141008   
stopwords                                                             0.324417   
class value                                                          -0.169477   

                                           hours  word count  char count  \
GPA                                    -0.198078    0.101102    0.110002   
1 - no, 2 - maybe, 3 - yes, 4 - intern  0.020963    0.339824    0.335874   
hours                                   1.000000   -0.122676   -0.113463   
word count                             -0.122676    1.000000    0.994536   
char count                             -0.113463    0.994536    1.000000   
avg_word                                0.030919   -0.117322   -0.042036   
stopwords                              -0.130189    0.992611    0.980561   
class value                            -0.183342   -0.183683   -0.203594   

                                        avg_word  stopwords  class value  
GPA                                     0.134431   0.092039    -0.156800  
1 - no, 2 - maybe, 3 - yes, 4 - intern -0.141008   0.324417    -0.169477  
hours                                   0.030919  -0.130189    -0.183342  
word count                             -0.117322   0.992611    -0.183683  
char count                             -0.042036   0.980561    -0.203594  
avg_word                                1.000000  -0.163788    -0.141155  
stopwords                              -0.163788   1.000000    -0.166921  
class value                            -0.141155  -0.166921     1.000000